### ISSUE TITLE
script-indent make me crazy!!!!!!!

### ISSUE THREAD

**Issue by wengpengfeijava** [vuejs/eslint-plugin-vue#362]
**Tell us about your environment**

* **ESLint Version:**  `^4.15.0`
* **eslint-plugin-vue Version:** `^4.0.0`
* **Node Version:** `^8.9.0`
* **IDE:** `webstorm 2017.2.5`

**Please show your full configuration:**

```javascript
// https://eslint.org/docs/user-guide/configuring
module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true
  },
  extends: [
    // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
    // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
    'plugin:vue/essential',
    // https://github.com/standard/standard/blob/master/docs/RULES-en.md
    'standard'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {
    // // 没找到解决方案， 一格式化webstorm script标签下面就会缩进，一缩进，eslint就报警。
    // // 只能从根源把他且了。
    // 'indent': ['off'],
    // allow async-await
    'generator-star-spacing': 'off',
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```vue
<template>
  <div id="app">
    
  </div>
</template>

<script>
 export default {
   name: 'App'
 }
</script>

```

**What did you expect to happen?**

```vue
<template>
  <div id="app">
    
  </div>
</template>

<script>
  // attension here. it is about 2 indent before javascript code
   export default {
     name: 'App'
   }
</script>

```

**What actually happened? Please include the actual, raw output from ESLint.**

This happend when i command in  `npm run dev`

```javascript
  ✘  http://eslint.org/docs/rules/indent  Expected indentation of 0 spaces but found 2  
  src\App.vue:10:1
    export default {
   ^

  ✘  http://eslint.org/docs/rules/indent  Expected indentation of 2 spaces but found 4  
  src\App.vue:11:1
      name: 'App'
   ^

  ✘  http://eslint.org/docs/rules/indent  Expected indentation of 0 spaces but found 2  
  src\App.vue:12:1
    }
   ^
```

As you can see above. This is not my expect things. Does any config to avoid these happen?

Otherwise, i may closing the indent config.although it is not a good experience.

**Comment by mysticatea** [comment]
Thank you for the report.

But it looks to be the errors of core [indent](https://eslint.org/docs/rules/indent) rule rather than our [vue/script-indent](https://github.com/vuejs/eslint-plugin-vue/blob/095ef4fb01e0ee96b616f2898946050b16e7769b/docs/rules/script-indent.md) rule.

If you use `vue/script-indent` rule with `baseIndent` is `1`, it would let you use one indentation in the `<script>` tags.

> Note the `vue/script-indent` rule in the current version has a bug around property shorthand.

**Comment by wengpengfeijava** [comment]
J just try to configuring rules as

```javascript
'vue/script-indent': ['warn', 2, {
  'baseIndent': 1
}]
```

```javascript
<script>
   // attension here. it is about 2 indent before javascript code
  export default {
     name: 'App'
   }
</script>
```

It still show the error 

```javascript
  ✘  http://eslint.org/docs/rules/indent  Expected indentation of 0 spaces but found 2  
  src\App.vue:10:1
    export default {
   ^
  ...
```

Instead, I did removed the indent before the javascript code.

It showed warnings below

```javascript
  ⚠  https://google.com/#q=vue%2Fscript-indent  Expected indentation of 2 spaces but found 0 spaces  
  src\App.vue:10:1
  export default {
   ^

  ⚠  https://google.com/#q=vue%2Fscript-indent  Expected indentation of 4 spaces but found 2 spaces  
  src\App.vue:11:1
    name: 'App'
   ^
```

I do really have no ideas about these , please help me!  (⊙o⊙)

**Comment by mysticatea** [comment]
```
'indent': 'off',
'vue/script-indent': ['warn', 2, {
  'baseIndent': 1
}]
```

**Comment by wengpengfeijava** [comment]
All right! It seems to be possible. thank you very much .

**Comment by wengpengfeijava** [comment]
I spent much time to know why dose these code warned me 

```javascript

const router = [];

new Vue({
     router
})
```

I awake to know that you've said 

> Note the vue/script-indent rule in the current version has a bug around property shorthand.

So , i am still  perplexed.

**Comment by mysticatea** [comment]
That bug has been fixed in #346, but we have not released it yet. Please wait for the next release...

Closing since it's fixed on `master` branch.

**Comment by michalsnik** [comment]
Sorry for late release.. Enjoy :) https://github.com/vuejs/eslint-plugin-vue/releases/tag/v4.2.1

**Comment by wengpengfeijava** [comment]
Thank you for your bug fixed.

**Comment by BarryZhan** [comment]
If you are using webstorm, you can `File => Setting => Editor => Code Style => HTML`  add `Do not indent children of `  rule to `script`

**Comment by leeuwd** [comment]
When working with `*.vue` files, this seems a better option to me: to disable the `indent` rule only for Vue files and set a base indent:

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  'extends': [
    'plugin:vue/strongly-recommended'
  ],
  rules: {
    'indent': ['error', 2],
    'vue/script-indent': [
      'error',
      2,
      { 'baseIndent': 1 }
    ],
  },
  'overrides': [
    {
      'files': ['*.vue'],
      'rules': {
        'indent': 'off'
      }
    }
  ]
};
```

**Comment by Grawl** [comment]
Just faced this problem

changed this:

```json
		"vue/script-indent": [
			"error",
			"tab",
			{
				"baseIndent": 1
			}
		]
```

to this:

```json
		"vue": {
			"script-indent": [
				"error",
				"tab",
				{
					"baseIndent": 1
				}
			]
		}
```

and problem solved

**Comment by cymonx** [comment]
> Just faced this problem
> 
> changed this:
> 
> ```json
> 		"vue/script-indent": [
> 			"error",
> 			"tab",
> 			{
> 				"baseIndent": 1
> 			}
> 		]
> ```
> 
> to this:
> 
> ```json
> 		"vue": {
> 			"script-indent": [
> 				"error",
> 				"tab",
> 				{
> 					"baseIndent": 1
> 				}
> 			]
> 		}
> ```
> 
> and problem solved

where is the place to changed. #vue_noob?

**Comment by begueradj** [comment]
@mysticatea 
I want to get indentation errors instead of warnings, I did this but I am still getting warnings instead:

```
rules: {                                                                                                                                       
    "vue/script-indent": ["error", 2, {                                                                                                          
      "baseIndent": 0,                                                                                                                           
      "switchCase": 0,                                                                                                                           
      "ignores": []                                                                                                                              
    }]                                                                                                                                           
  }  
```
What am I missing? Thank you

**Comment by apoorvpatne10** [comment]
Why not just include `/* eslint-disable */` to ignore all warnings? Worked for me.

**Comment by Grawl** [comment]
this is not best solution

**Comment by apoorvpatne10** [comment]
I never said it's the best solution. It seems to work well for me, and might work for others too.

**Comment by Grawl** [comment]
ofcourse it works

**Comment by Ezra-Siton-UIX** [comment]
Why this issue is closed? Mega bug - still happens.

**Comment by Tofandel** [comment]
@apoorvpatne10 You might as well remove eslint if you need to disable it in all your files

@begueradj You need to use the indent rule I think

`'indent': ['error', 2],`

**Comment by MaximSemenov** [comment]
If you are using WebStorm or PhpStrom by JetBrains open setting by 

```Cmd + ' , ' (on Mac)```

Editor->Code Style -> Javascript

Select checkbox `Use tab character'`

![Screenshot 2020-05-23 at 13 42 35](https://user-images.githubusercontent.com/20158607/82728788-49f8fc00-9cfb-11ea-855c-cb99da46840b.png)

**Comment by SmelayaPanda** [comment]
> If you are using webstorm, you can `File => Setting => Editor => Code Style => HTML` add `Do not indent children of ` rule to `script`
@BarryZhan 

For last Webstorm versions `File => Setting => Editor => Code Style => Vue` add `Indent children of top level tag: `script, style`

**Comment by manojbaishya** [comment]
This bug still exists when using [Volar](https://github.com/johnsoncodehk/volar) and format document.

`.eslintrc.cjs`
```javascript
/* eslint-env node */
module.exports = {
    root: true,
    extends: [
        "eslint:recommended",
        'plugin:vue/base',
        'plugin:vue/vue3-essential',
        'plugin:vue/vue3-recommended',
        'plugin:vue/vue3-strongly-recommended',
        '@vue/typescript/recommended',
        "@vue/eslint-config-typescript/recommended",
    ],
    rules: {
        "indent": ["error", 4],
        "vue/html-indent": ["error", 4, {
            "attribute": 1,
            "baseIndent": 1,
            "closeBracket": 0,
            "alignAttributesVertically": true,
            "ignores": []
        }],
        "vue/script-indent": ["error", 4, {
            "baseIndent": 1,
            "switchCase": 1,
            "ignores": []
        }],
    },
    overrides: [
        {
            files: ['*.vue'],
            rules: {
                indent: 'off'
            }
        }
    ]
};

```

### Before fixing the problems:

![image](https://user-images.githubusercontent.com/28330014/194739027-41ce1c5c-830a-4097-9dd0-c3ea25963d8c.png)

### After fixing using Code Actions:
![image](https://user-images.githubusercontent.com/28330014/194739042-b64e86b0-a006-4a6b-8772-3aec594a1e05.png)
![image](https://user-images.githubusercontent.com/28330014/194739055-f5fadb38-6062-4fc3-b83e-6912b54e3638.png)

### Using Volar Document Formatter -> Back to the same problem!
![image](https://user-images.githubusercontent.com/28330014/194739075-bd0bab87-7ed4-4dc3-9b3b-0459751cb44f.png)
![image](https://user-images.githubusercontent.com/28330014/194739094-d415e761-f312-40e3-8170-49ca44faba7b.png)

Any way to make Volar compatible with the “vue/script-indent” rule?
